From 5a4730e94a15c14c7f430efd1d30135cbdc3bb3c Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Wed, 12 Feb 2020 07:31:57 -0600
Subject: [PATCH 10/21] arm64: dts: fsl-imx8mn: Enable dispmix_gpr

The DSI and eLCDIF interfaces are expecting a reference to dispmix_gpr
which is present in the i.MX8M Mini device tree, but not the i.MX8M Nano.

This patch enables the dispmix_gpr for the i.MX8M Nano which will allow
video support on the i.MX8M Nano.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mn.dtsi | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mn.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mn.dtsi
index f68c8913a6c5..9926bbd0c275 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mn.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mn.dtsi
@@ -963,6 +963,7 @@
 					 <&clk IMX8MN_SYS_PLL1_800M>;
 		assigned-clock-rate = <594000000>, <500000000>, <200000000>;
 		interrupts = <GIC_SPI 5 IRQ_TYPE_LEVEL_HIGH>;
+		lcdif-gpr = <&dispmix_gpr>;
 		resets = <&lcdif_resets>;
 		power-domains = <&dispmix_pd>;
 		status = "disabled";
@@ -990,6 +991,7 @@
 					 <&clk IMX8MN_VIDEO_PLL1_OUT>;
 		assigned-clock-rates = <266000000>, <594000000>;
 		interrupts = <GIC_SPI 18 IRQ_TYPE_LEVEL_HIGH>;
+		dsi-gpr = <&dispmix_gpr>;
 		resets = <&mipi_dsi_resets>;
 		power-domains = <&mipi_pd>;
 		status = "disabled";
@@ -1059,6 +1061,11 @@
 		reg = <0x0 0x32e28060 0x0 0x28>;
 	};
 
+	dispmix_gpr: display-gpr@32e28000 {
+		compatible = "fsl, imx8mm-iomuxc-gpr", "syscon";
+		reg = <0x0 0x32e28000 0x0 0x100>;
+	};
+
 	dispmix-reset {
 		compatible = "simple-bus";
 		#address-cells = <2>;
-- 
2.17.1

